department-of-veterans-affairs/vets-website
# [CST] Add the ability to mock fetching a list of claims

## Original Issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/42401

## Background
When attempting to view a list of claims in the claims status tool while doing local development, the api call made to retrieve the list of claims errors. This makes it impossible to view the list unless the developer either mocks the data in `vets-api` or `vets-website`. In `vets-website` we already have a set of fixtures that we use for testing; I think that it makes sense to utilize those fixtures to solve this problem.

## Build notes
* You should not need to use a specific staging user for this to work

## Steps to test
* Log in with any staging user
* Navigate to `localhost:3001/track-claims/your-claims`
* Scroll down to the section labeled `Your claims or appeals`
* You should see a list containing 2 claims
* Claims can be differentiated from appeals by the heading text. It should be `Claim for ...` for claims

## Code changes
* Adding a `USE_MOCKS` boolean that should be `true` when developing locally and `false` when in dev, staging and production
* Importing the `claims-list.json` file from our fixtures folder
* In `getClaimsV2`, we are using the `claims-list.json` content as part of a call to `fetchClaimsSuccess` in the `onError` callback. **Note:** Since the call to `vets-api` will always fail, we will end up in the `onError` callback

## How was your code tested
Tests continue to pass

## Acceptance criteria
- [x] Visiting `/track-claims/your-claims` should display a list of mocked claims
